lib/ansible/plugins/inventory/foreman.py

```diff
@@ -219,7 +224,14 @@ def _populate(self):
 
                 # set host vars from params
                 if self.get_option('want_params'):
-                    for p in self._get_all_params_by_id(host['id']):
+
+                    # handle legacy foreman which is missing all_parameters api option
+                    if host.get('all_parameters'):
```

Comment:
When `all_parameters` is present but empty, it will also do an API request. Untested, but how about:
```python
try:
    host_params = host['all_parameters']
except KeyError:
    host_params = self._get_all_params_by_id(host['id'])
```

Reply:
@ekohl Good catch.  Just tested and that will be sufficient for this.